src/components/Common/SearchBarDropdown.tsx

```diff
@@ -0,0 +1,207 @@
+import SearchRoundedIcon from "@mui/icons-material/SearchRounded";
+import ClearIcon from "@mui/icons-material/Clear";
+import {
+  Autocomplete,
+  Avatar,
+  Box,
+  IconButton,
+  InputAdornment,
+  ListItemAvatar,
+  ListItemButton,
+  ListItemText,
+  TextField,
+  Typography,
+  debounce,
+} from "@mui/material";
+import { useEffect, useMemo, useRef, useState } from "react";
+import axios from "axios";
+import { useAuth0 } from "@auth0/auth0-react";
+import { SelectedUser } from "../../state/slices/messagesSlice";
+import { useNavigate } from "react-router-dom";
+
+type SearchBarDropDownProps = {
+  placeholder: string;
+};
+
+const styles = {
+  autocomplete: {
+    "&.MuiAutocomplete-hasClearIcon .MuiOutlinedInput-root": { paddingX: 1 },
+    position: "relative",
+    padding: 0,
+  },
+  box: {
+    paddingBottom: 1,
+    paddingTop: 0,
+    paddingX: 2,
+  },
+  listBox: {
+    maxHeight: "60vh",
+  },
+  searchIcon: {
+    paddingRight: 0,
+    "&.Mui-disabled": {
+      color: "gray.main",
+    },
+  },
+  searchIconFocused: {
+    paddingRight: 0,
+    "&.Mui-disabled": {
+      color: "primary.main",
+    },
+  },
+};
+
+const SearchBarDropDown = ({ placeholder }: SearchBarDropDownProps) => {
+  const [focusSearchBar, setFocusSearchBar] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [keywords, setKeywords] = useState("");
+  const [loading, setLoading] = useState(false);
+  const [searchOptions, setSearchOptions] = useState<SelectedUser[]>([]);
+  const { getAccessTokenSilently } = useAuth0();
+  const navigate = useNavigate();
+  const debouncedFetch = useMemo(
+    () => debounce((keywords: string) => fetchUsers(keywords), 100),
+    []
+  );
+
+  const fetchUsers = async (keywords: string) => {
+    setLoading(true);
+    if (keywords.length === 0) {
+      return;
+    }
```

Comment:
In this code path `loading` never gets set back to false. Also, can we combine with the `keywords.trim() === ""` code path on line 89? Feels like they're doing the same thing

Reply:
Alright so I've moved the logic for `keywords.trim() ==== ""` to the if statement before the try catch block, and also setting loading to false inside of that.